tests/phpunit/tests/rest-api/rest-pattern-directory-controller.php

```diff
@@ -310,6 +340,174 @@ static function( $response ) {
 		$this->assertSame( 'modified the cache', $patterns[0] );
 	}
 
+	/**
+	 * Tests if the provided query args are passed through to the wp.org API.
+	 *
+	 * @dataProvider data_get_items_query_args
+	 *
+	 * @covers WP_REST_Pattern_Directory_Controller::get_items
+	 *
+	 * @since 6.2.0
+	 *
+	 * @param string $param    Query parameter name (ex, page).
+	 * @param mixed  $value    Query value to test.
+	 * @param bool   $is_error Whether this value should error or not.
+	 * @param mixed  $expected Expected value (or expected error code).
+	 */
+	public function test_get_items_query_args( $param, $value, $is_error, $expected ) {
+		wp_set_current_user( self::$contributor_id );
+		add_filter( 'pre_http_request', array( $this, 'mock_request_to_apiwporg_url' ), 10, 3 );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/pattern-directory/patterns' );
+		if ( $value ) {
+			$request->set_query_params( array( $param => $value ) );
+		}
+
+		$response = rest_do_request( $request );
+		$data     = $response->get_data();
+		if ( $is_error ) {
+			$this->assertSame( $expected, $data['code'], 'Response error code does not match' );
+			$this->assertStringContainsString( $param, $data['message'], 'Response error message does not match' );
+		} else {
+			$this->assertCount( 1, self::$http_request_urls, 'The number of HTTP Request URLs is not 1' );
+			$this->assertStringContainsString( $param . '=' . $expected, self::$http_request_urls[0], 'The param and/or value do not match' );
+		}
+	}
+
+	/**
+	 * Data provider.
```

Comment:
```suggestion
	 * Data provider for test_get_items_query_args().
```

Reply:
I think this is unnecessary. The naming convention of `test_` and `data_` align to know that this data provider is for the matching test.

Reply:
Whoops forgot to share where the naming convention is identified the handbook 🤦‍♀️ 

>A [data provider](https://phpunit.readthedocs.io/en/9.5/writing-tests-for-phpunit.html#data-providers) is a secondary function which is linked to a test using a` @dataProvider `annotation in the docblock of the test method and is generally named after the test, replacing the test prefix in the method name with data.

https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#repetitive-tests.